### PR TITLE
move analysis out of spec package

### PR DESCRIPTION
**Pull request by casualjim**
refactoring as part of #420 

This step extracts analysis from the spec document
Moves yaml into its own packages for opt-in behavior
Moves loading into its own package

### CODE REVIEW

File: analysis/analyzer.go [hunk 1]
```diff
@@ -0,0 +1,621 @@
...
+	allRefs    map[string]spec.Ref
+	referenced struct {
+		schemas    map[string]SchemaRef
+		responses  map[string]*spec.Response
+		parameters map[string]*spec.Parameter
+	}
+}
+
+func (r *referenceAnalysis) addRef(key string, ref spec.Ref) {
+	if r.allRefs == nil {
```
Comment (vburenin) [inline analysis/analyzer.go]: Is there any particular reason to initialize it like this?

File: analysis/analyzer.go [hunk 2]
```diff
@@ -339,7 +353,7 @@ func (s *specAnalyzer) SecurityDefinitionsFor(operation *Operation) map[string]S
 }
 
 // ConsumesFor gets the mediatypes for the operation
-func (s *specAnalyzer) ConsumesFor(operation *Operation) []string {
+func (s *Spec) ConsumesFor(operation *spec.Operation) []string {
 	cons := make(map[string]struct{})
```
Comment (vburenin) [inline analysis/analyzer.go]: This might make sense to specify a capacity since it is already known.

File: spec/spec.go [hunk 3]
```diff
@@ -147,20 +147,20 @@ func New(data json.RawMessage, version string) (*Document, error) {
...
-		specAnalyzer: specAnalyzer{
-			spec:        spec,
-			consumes:    make(map[string]struct{}),
-			produces:    make(map[string]struct{}),
-			authSchemes: make(map[string]struct{}),
-			operations:  make(map[string]map[string]*Operation),
-			allSchemas:  make(map[string]SchemaRef),
-			allOfs:      make(map[string]SchemaRef),
-		},
+		// specAnalyzer: specAnalyzer{
```
Comment (vburenin) [inline spec/spec.go]: Do we need this commented code?

**Comment by casualjim** [comment]
lgtm

**Pull request merged**